common/interfaces/user_interface/ui_joystick_training.c

```diff
@@ -0,0 +1,227 @@
+ *****************************************************************************/
+
+#include "ui_joystick_training.h"
+
+#include "ui_events_priv.h"
+#ifdef DEV_BUILD
+#include "dev_utils.h"
+#endif
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/**
+ * @brief struct to store message data
+ * @details
+ *
+ * @see
+ * @since v1.0.0
+ *
+ * @note Add constant
+ */
+struct Message_Data {
+  char *message;
+};
+
+/**
+ * @brief struct to store message lvgl object
+ * @details
+ *
+ * @see
+ * @since v1.0.0
+ *
+ * @note
+ */
+struct Message_Object {
+  lv_obj_t *message;
+};
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/**
+ * @brief Create message screen
+ * @details
+ *
+ * @see
+ * @since v1.0.0
+ */
+static void joystick_train_create();
+
+/**
+ * @brief Clear screen
+ * @details
+ *
+ * @param
+ *
+ * @return
+ * @retval
+ *
+ * @see
+ * @since v1.0.0
+ *
+ * @note
+ */
+static void joystick_train_destructor();
+
+/**
+ * @brief Next button event handler.
+ * @details
+ *
+ * @param next_btn Next button lvgl object.
+ * @param event Type of event.
+ *
+ * @return
+ * @retval
+ *
+ * @see
+ * @since v1.0.0
+ *
+ * @note
+ */
+static void next_btn_event_handler(lv_obj_t *obj, const lv_event_t event);
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+static struct Message_Data *data = NULL;
+static struct Message_Object *obj = NULL;
+static joystick_actions_t action = JS_ACTION_CENTER;
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+
+static void joystick_train_destructor() {
+  if (data != NULL) {
+    memzero(data, sizeof(struct Message_Data));
+    free(data);
+    data = NULL;
+  }
+  if (obj != NULL) {
+    free(obj);
+    obj = NULL;
+  }
+}
+
+static void next_btn_event_handler(lv_obj_t *obj, const lv_event_t event) {
+  if (event == LV_EVENT_KEY && action == lv_indev_get_key(ui_get_indev())) {
+    ui_set_confirm_event();
```

Comment:
Shouldn't we trigger delete event from here?

Reply:
this is intentional and is aligned w.r.t. #259 

1. the LVGL events handling appears unsafe for deleted object references.
2. from inside of event handlers leads to recursive event handler calls (as the `lv_obj_clean()` does in-place object deletion).